src/sema/diagnostics.rs

```diff
@@ -5,48 +5,43 @@ use codespan_reporting::{diagnostic, files, term};
 use serde::Serialize;
 use std::{io, sync::Arc};
 
-/// Print the diagnostics to stderr with fancy formatting
-pub fn print_diagnostics(cache: &FileResolver, ns: &Namespace, debug: bool) {
-    let (files, file_id) = convert_files(ns, cache);
-
-    let writer = term::termcolor::StandardStream::stderr(term::termcolor::ColorChoice::Always);
-    let config = term::Config::default();
-
-    for msg in &ns.diagnostics {
-        if msg.level == Level::Debug && !debug {
-            continue;
-        }
-
-        let diagnostic = convert_diagnostic(msg, &file_id);
-
-        term::emit(&mut writer.lock(), &config, &files, &diagnostic).unwrap();
-    }
+#[derive(Default)]
+pub struct Diagnostics {
+    pub contents: Vec<Diagnostic>,
+    has_error: bool,
 }
 
-/// Print the diagnostics to stdout with plain formatting
-pub fn print_diagnostics_plain(cache: &FileResolver, ns: &Namespace, debug: bool) {
-    let (files, file_id) = convert_files(ns, cache);
+impl Diagnostics {
+    pub fn any_errors(&self) -> bool {
+        self.has_error
+    }
 
-    let config = term::Config::default();
+    /// Borrow the contents vector
+    /// We should not modify the contents without this data strcuture
+    pub fn get_contents(&self) -> &Vec<Diagnostic> {
+        &self.contents
+    }
 
-    for msg in &ns.diagnostics {
-        if msg.level == Level::Debug && !debug {
-            continue;
+    pub fn add_diagnostic(&mut self, diagnostic: Diagnostic) {
```

Comment:
you're repeating diagnostic in the function name. We already know it's a diagnostic. Please just `fn add()`

Reply:
maybe `fn push()` would be even better, keeps the diff small

Reply:
maybe `fn push()` would be even better, keeps the diff small